src/context.c

```diff
@@ -140,6 +140,43 @@ static gchar* get_cmdline_bootname(void)
 	return bootname;
 }
 
+static gchar* get_custom_bootname(void)
```

Comment:
This could be a GError method. Alternatively, at least emit `g_warning()` messages for errors inside this function.

Reply:
I decide for keeping the method simple as the `get_cmdline_bootname`, so it simply outputs the errors and returns NULL when bootname is not found.

Reply:
Honestly, on the g_warning, I think it modifies the behaviour too much: when running tests, if there is a g_warning, the test bail out and abort instead of continuing and complaining about an assertion failed.
I think we must stay with g_message here.

Reply:
> Honestly, on the g_warning, I think it modifies the behaviour too much

The tests are configured to handle unexpected warnings as fatal errors.
We normally emit a g_message() if the error is expected and just for information purposes.

It may be a question of whether we expect the custom backend script to return the 'get-current' information or not. If it is actually seen as optional you are probably right that a `g_message()` would be sufficient.

Reply:
Well, if we want to keep the same approach of `get_cmdline_bootname`, that function can fail and return null without any warning.
I would maintain the same approach, the custom handler could fail without any warning and return NULL.